upload-api/tables/index.js

```diff
@@ -26,3 +26,65 @@ export const uploadTableProps = {
   // space + root must be unique to satisfy index constraint
   primaryIndex: { partitionKey: 'space', sortKey: 'root' },
 }
+
+/** @type TableProps */
+export const delegationTableProps = {
+  fields: {
+    cid: 'string',        // `baf...x`
+    audience: 'string',   // `did:web:service`
+    issuer: 'string',     // `did:key:agent`
+    expiration: 'string', // `9256939505` (unix timestamp)
+    insertedAt: 'string', // `2022-12-24T...`
+    updatedAt: 'string',  // `2022-12-24T...`
+  },
+  // TODO does this index setup seem right?
+  // we want to query by audience, but that won't necessarily be unique, so use cid as sortKey
+  primaryIndex: { partitionKey: 'audience', sortKey: 'cid' },
+}
+
+/** @type TableProps */
+export const provisionTableProps = {
+  fields: {
+    cid: 'string',        // `baf...x` (CID of invocation that created this provision)
+    consumer: 'string',   // `did:key:space` (DID of the actor that is consuming the provider, e.g. a space DID)
+    provider: 'string',   // `did:web:service` (DID of the provider, e.g. a storage provider)
+    customer: 'string',    // `did:key:agent` (DID of the actor that authorized this provision)
+    insertedAt: 'string', // `2022-12-24T...`
+    updatedAt: 'string',  // `2022-12-24T...`
+  },
+  // TODO does this index setup seem right?
+  // we want to query by consumer, and I think it should be unique?
+  // TODO also are we sure that a single invocation will only create a single provision? the D1 schema used CID as a PRIMARY KEY so I think this should be fine, but will it always be true?
+  primaryIndex: { partitionKey: 'consumer' },
+}
+
+/** VS */
+
+
+/** @type TableProps */
+export const subscriptionTableProps = {
+  fields: {
+    cause: 'string',        // `baf...x` (CID of invocation that created this provision)
+    provider: 'string',   // `did:web:service` (DID of the provider, e.g. a storage provider)
+    customer: 'string',   // `did:mailto:agent` (DID of the user account)
+    order: 'string',      // string (arbitrary string associated with this subscription)
+    insertedAt: 'string', // `2022-12-24T...`
+    updatedAt: 'string',  // `2022-12-24T...`
+  },
+  // TODO does this index setup seem right?
+  primaryIndex: { partitionKey: 'provider', sortKey: 'order' },
+}
+
+/** @type TableProps */
+export const consumerTableProps = {
+  fields: {
+    cause: 'string',        // `baf...x` (CID of invocation that created this provision)
+    consumer: 'string',   // `did:key:space` (DID of the actor that is consuming the provider, e.g. a space DID)
+    provider: 'string',   // `did:web:service` (DID of the provider, e.g. a storage provider)
+    order: 'string',      // string (arbitrary string associated with this subscription)
+    insertedAt: 'string', // `2022-12-24T...`
+    updatedAt: 'string',  // `2022-12-24T...`
+  },
+  // TODO does this index setup seem right?
+  primaryIndex: { partitionKey: 'provider', sortKey: 'order' },
```

Comment:
maybe we want the consumer as `sortKey`? we need to flush out what kind of queries we need here

Reply:
yea - I've started to do that a bit in https://github.com/web3-storage/w3infra/pull/194/files#diff-924a94ba76534a93c7fa4c89f8e70037f21694b6374beaca7a2c6ebf3fdb2006 - I've changed the partition and sort keys for these tables - definitely don't want to use provider as it will be one of two values for a while...